programs/clearing_house/src/math/repeg.rs

```diff
@@ -428,6 +437,230 @@ pub fn adjust_peg_cost(
     Ok((market_clone, cost))
 }
 
+pub fn adjust_amm(
+    market: &Market,
+    optimal_peg: u128,
+    budget: u128,
+    adjust_k: bool,
+) -> ClearingHouseResult<(Market, i128)> {
+    let curve_update_intensity = cast_to_i128(min(market.amm.curve_update_intensity, 100_u8))?;
+    // let current_peg = market.amm.peg_multiplier;
+
+    // return early
+    if optimal_peg == market.amm.peg_multiplier || curve_update_intensity == 0 {
+        // msg!("RETURN adjust_prepeg EARLY");
+        return Ok((*market, 0));
+    }
+
+    let delta_peg = cast_to_i128(optimal_peg)?
+        .checked_sub(cast_to_i128(market.amm.peg_multiplier)?)
+        .ok_or_else(math_error!())?;
+
+    let mut per_peg_cost = cast_to_i128(market.amm.quote_asset_reserve)?
```

Comment:
do you underestimate or overestimate the peg cost by shaving the reserve precision to 1e3 here?